frontend/public/components/row-filter.jsx

```diff
@@ -76,6 +78,29 @@ class CheckBoxes_ extends React.Component {
     }
   }
 
+  selectAll(event) {
+    const selected = _.map(this.props.items, 'id');
+    this.setState({selected}, () => this.applyFilter());
+  }
+
+  countNumItems() {
+    const selectedCheckboxes = _.map(this.state.selected, (id) => {
+      return this.props.numbers[id] || 0;
+    });
+    const numSelectedItems = _.reduce(selectedCheckboxes, (n, sum) => {
+      return n ? sum + n : sum;
+    }, 0);
+    const numTotalItems = _.reduce(this.props.numbers, (n, sum) => {
+      return n ? sum + n : sum;
+    }, 0);
+    const allSelected = _.isEmpty(_.xor(this.state.selected, _.map(this.props.items, 'id')));
+    if (allSelected) {
+      return numTotalItems === 1 ? `1 Item` : `${numTotalItems} Items`;
```

Comment:
Use `pluralize` to handle plurals.

```suggestion
      return pluralize(numTotalItems, 'Item');
```